src/connector/src/source/datagen/source/generator.rs

```diff
@@ -57,24 +57,36 @@ impl DatagenEventGenerator {
         loop {
             // generate `partition_rows_per_second` rows per second
             interval.tick().await;
-            let mut msgs = vec![];
-            for _ in 0..self.partition_rows_per_second {
-                let value = Value::Object(
-                    self.fields_map
+            let mut count = 0;
+            let max_row = 1024;
+            while count < self.partition_rows_per_second {
+                let mut msgs = vec![];
+                while msgs.len() < max_row {
+                    let data = self
+                        .fields_vec
                         .iter_mut()
-                        .map(|(name, field_generator)| {
-                            (name.to_string(), field_generator.generate(self.offset))
-                        })
-                        .collect(),
-                );
-                msgs.push(SourceMessage {
-                    payload: Some(Bytes::from(value.to_string())),
-                    offset: self.offset.to_string(),
-                    split_id: self.split_id.clone(),
-                });
-                self.offset += 1;
+                        .map(|field_generator| field_generator.generate_datum(self.offset))
+                        .collect();
+                    let value = unsafe {
+                        let row = OwnedRow::new(data);
+                        let row = Box::new(row);
+                        let ptr = Box::into_raw(row);
```

Comment:
Cannot use `Box::leak -> &mut T` instead of `Box::into_raw -> *mut T`. 
Even after `Box::from_raw`, it still leaks...

Probably has something to do with Box itself's memory layout?
But `leak` -> `from_raw` is suggested in `Box`'s doc.